### PR TITLE
display total length and running time in simulation tabs

### LINKED ISSUE #9678: lmr: front: display total length and running time in simulation tabs
# Description

Users requested to have a quick overview of the total length and duration for the different simulation.

# AC

- [ ] This UI has been implemented in the simulation tabs, with total length and trip duration under the date and time of the simulation

Image

### PR DESCRIPTION
**Pull request by kmer2016**
closes #9678

### CODE REVIEW

**Comment by theocrsb** [comment]
Shouldn't the text be on one line?

mockup: 
image

app: 
image

**Comment by kmer2016** [comment]
> Shouldn't the text be on one line?
> 
> mockup: image
> 
> app: image

Good catch! It seems this issue was introduced during the rebase !

**Comment by theocrsb** [comment]
I think a small 10px margin is missing
mockup:
image

app: 
image

**Review by theocrsb — APPROVED** [review]
LGTM and tested

File: front/public/locales/en/translation.json [hunk 1]
```diff
@@ -50,6 +50,11 @@
         "title": "Failed to save modifications",
         "text": "Failed to publish your modifications"
       }
+    },
+    "units": {
+      "day": "j",
```
Comment (SharglutDev) [inline front/public/locales/en/translation.json]: You reversed the en and fr trad :)

File: front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx [hunk 2]
```diff
@@ -53,40 +54,64 @@ const StdcmSimulationNavigator = ({
...
-              <div className="simulation-name">
-                <div>
-                  {outputs && !hasConflicts(outputs)
-                    ? t('simulationName.withOutputs', { id })
-                    : t('simulationName.withoutOutputs')}
+          {simulationsList?.map(({ id, creationDate, outputs }, index) => {
+            let formatedTotalLength = '';
+            let formatedTripDuration = '';
+
+            if (hasResults(outputs) && outputs.results.simulation.final_output.times.length > 0) {
```
Comment (SharglutDev) [inline front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx]: If we have an output with result we are sure we will have at least 2 number in times no ?

File: front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx [hunk 3]
```diff
@@ -53,40 +54,64 @@ const StdcmSimulationNavigator = ({
...
+              const { results } = outputs;
+              const lastPointTime = results.simulation.final_output.times.at(-1)!;
+              const departureTimeInTs = new Date(results.departure_time).getTime();
```
Comment (SharglutDev) [inline front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx]:
```suggestion
              const departureTimeInMs = new Date(results.departure_time).getTime();
```

File: front/public/locales/fr/translation.json [hunk 4]
```diff
@@ -50,6 +50,11 @@
         "title": "Echec de l'enregistrement des modifications",
         "text": "Vos modifications n'ont pas été publiées"
       }
+    },
+    "units": {
+      "day": "d",
+      "hour": "h",
+      "minute": "min"
```
Comment (SharglutDev) [inline front/public/locales/fr/translation.json]: Do we want the trad for these if they are the same between fr and en ? If yes, we should also add `km`

File: front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx [hunk 5]
```diff
@@ -27,7 +28,7 @@ const StdcmSimulationNavigator = ({
   isCalculationFailed,
   onSelectSimulation,
 }: StdcmSimulationNavigatorProps) => {
-  const { t } = useTranslation('stdcm', { keyPrefix: 'simulation.results' });
+  const { t } = useTranslation();
```
Comment (SharglutDev) [inline front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx:31]: Why did you change this ?

File: front/src/styles/scss/applications/stdcm/_results.scss [hunk 6]
```diff
@@ -75,13 +84,15 @@
...
+              margin-left: 8px;
               color: rgb(60, 202, 128);
-              padding-left: 8px;
             }
           }
 
           &.selected {
-            .creation-date {
+            .simulation-metadata {
               color: rgb(121, 118, 113);
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/stdcm/_results.scss]: Not coming from your work but 

```suggestion
              color: var(--grey50);
```

File: front/src/styles/scss/applications/stdcm/_results.scss [hunk 7]
```diff
@@ -97,7 +108,8 @@
           }
 
           &.retained {
-            .simulation-name {
+            .simulation-name,
+            .simulation-metadata {
               color: rgb(11, 114, 60);
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/stdcm/_results.scss]: Same here

```suggestion
              color: var(--success60);
```

File: front/src/styles/scss/applications/stdcm/_results.scss [hunk 8]
```diff
@@ -51,12 +51,11 @@
         overflow-x: hidden;
 
         .simulation-item {
-          width: 125px;
           display: inline-block;
-          margin: 9px 64px 0 0;
+          margin: 9px 39px 0 0;
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/stdcm/_results.scss]: Not your fault, but there is no margin-top here

image

```suggestion
          margin-right: 39px;
```

File: front/src/styles/scss/applications/stdcm/_results.scss [hunk 9]
```diff
@@ -97,7 +108,8 @@
           }
 
           &.retained {
-            .simulation-name {
+            .simulation-name,
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/stdcm/_results.scss:112]: Still not your fault but when a simulation is retained, the margin right is not the same

image

```suggestion
            margin-right: 29px;
            
            .simulation-name,
```

File: front/src/styles/scss/applications/stdcm/_results.scss
Comment (SharglutDev) [inline front/src/styles/scss/applications/stdcm/_results.scss:1]: On screen with a resolution less than ~1390px, it goes like this : 

image

**Review by SharglutDev — COMMENTED** [review]
Thank you for the PR, left a few comments

File: front/src/styles/scss/applications/stdcm/_results.scss (hunk 8 above)
Comment (SharglutDev) [reply front/src/styles/scss/applications/stdcm/_results.scss]: Ok just had a call with @thibautsailly. Turns out that the box of the navigator header should always have a width of 160px. The underline was selected/retained should follow this width.

And the gap between each header should be 32px (mock up should have been updated).

Sorry for these requests but I think it's a good moment to update this since you are updating this component.

File: front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx (hunk 2 above)
Comment (kmer2016) [reply front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx]: Good question, I wasn't sure, so I added the check to be safe.

File: front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx (hunk 5 above)
Comment (kmer2016) [reply front/src/applications/stdcm/components/StdcmResults/StdcmSimulationNavigator.tsx:31]: I initially kept it this way because I was translating duration units directly in the component and they are in another namespace, but I later moved the translation inside the formatting function.

Now, I’m leaving it as is because I added 'Km' to the translation dictionary.

File: front/public/locales/fr/translation.json [hunk 10]
```diff
@@ -50,6 +50,12 @@
         "title": "Echec de l'enregistrement des modifications",
         "text": "Vos modifications n'ont pas été publiées"
       }
+    },
+    "units": {
+      "day": "j",
+      "hour": "h",
+      "minute": "min",
+      "km": "km"
```
Comment (SharglutDev) [inline front/public/locales/fr/translation.json]: For alphabetical order :) (same for en)

```suggestion
      "km": "km",
      "minute": "min"
```

File: front/src/styles/scss/applications/stdcm/_results.scss
Comment (SharglutDev) [reply front/src/styles/scss/applications/stdcm/_results.scss:1]: can't reproduce anymore

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, good job ! You can resolve the last comment when fixed and merge :)
